Yoast/javascript
# Create DurationInput component

## Summary


This PR can be summarized in the following changelog entry:

* Adds a DurationInput component, which can handle the input of a duration in hours, minutes, and seconds.

## Relevant technical choices:

* Cherry-picked https://github.com/Yoast/javascript/commit/6dc7613b484c9d6a116924e1a7e38562ee00fb1d because I was rummaging in the `ReactifiedComponentsWrapper` which has been split in that commit. Then decided I was overcomplicating the PR and reverted the cherry-pick. So that's what that mess is all about.
* I really wanted to use React Hooks, which is fine in our plugin, but was somehow broken in the components app. That's why, to test this PR, you will need to merge (*without committing*) another PR where I "fix" the components app: https://github.com/Yoast/javascript/pull/1000.
* Deviates from design slightly, checked for approval from UX: https://yoast.slack.com/archives/C01FYN1QGB1/p1610026441033700 

## Test instructions

This PR can be tested by following these steps:

First you'll need to get the components app working:
* `git fetch`
* `git merge fix-component-app-react-hooks-by-removing-hot-loader  --no-commit`
* This will give you a merge commit, but you only have to `accept incoming change` in the ReactifiedComponentsWrapper.
* In the root of the monorepo, run `yarn install --check-files` for good measure.
* Go to `apps/components` and run `yarn start`.
* Go to localhost:3333 and checkout the `Inputs` section on the `Reactified Components` page. There is a `Duration` input there. See that it works as expected:

• No negative durations.
• No minutes and seconds not higher than 59
• Keyboard input only accepts numbers. Arrow buttons should work.

⚠️ unfortunately the fact that we use `useState` here does make the merging of #1000 a requirement for this issue.

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  *

## UI changes

## Quality assurance


Fixes P1-292
